### ISSUE TITLE
Unable to open link in plex-desktop

### ISSUE THREAD

**Issue by George-Miao** [NixOS/nixpkgs#349158]
### Describe the bug
When clicking on any links in `plex-desktop`, an error occurred saying 
```
XPCOMGlueLoad error for file /nix/store/9mi39p4nkj66lv5y4qyy09kxdfpgrb1d-firefox-130.0/lib/firefox/libmozgtk.so:
/nix/store/cdnwvy5zyh6la8x1cal00xmvsj8x3dai-wayland-1.23.1/lib/libwayland-cursor.so.0: undefined symbol: wl_proxy_marshal_flags
Couldn't load XPCOM.
```

### Steps To Reproduce
1. open `plex-desktop` from unstable nixpkgs in command line
2. click on sign in at top right of the app
3. click on sign in
4. nothing happens, an error was printed

### Expected behavior
Sign in/sign up link should be opened in browser.

### Notify maintainers

@detroyejr 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.50, NixOS, 24.11 (Vicuna), 24.11.20240911.4f807e8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - nixpkgs: `/nix/store/2pw5bsq1r80lcvvh2wp89m142nln1zpl-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by detroyejr** [comment]
There have been a few issues that look similar to this. Could you let me know if the solutions in #342824 or #341968 solve this?

The error message looks a little different though so these may not help here.

**Comment by George-Miao** [comment]
> There have been a few issues that look similar to this. Could you let me know if the solutions in #342824 or #341968 solve this? Adding `xdg.portal.xdgOpenUsePortal` has usually helped with links not working.

Thanks for responding. I tried using `xdg.portal.xdgOpenUsePortal` but no luck. Still the same error. I'm using hyprland.

**Comment by detroyejr** [comment]
```
  xdg.portal = {
    enable = true;
    extraPortals = [ pkgs.xdg-desktop-portal-gtk ];
    xdgOpenUsePortal = true;
  };
```
Here's the full snippet I use to make links work on Hyprland. You may need to reboot for the change to take effect.

It does seem like this is something else though, I'll see if I can find out why this might be happen.

**Comment by detroyejr** [comment]
Following up on this. It seems to be a common problem with these types of applications (see https://github.com/NixOS/nixpkgs/issues/200833 and https://github.com/NixOS/nixpkgs/issues/160923 for some very long discussions on this topic). If the above doesn't solve it for you, I'd be curious if you're mixing stable/unstable package versions anywhere?

My setup is similar to yours, but I'm not able to reproduce.

**Comment by detroyejr** [comment]
Yeah, nothing looks obviously off to me. Testing without the snippet I mentioned earlier does give me the same error but that goes away as soon as I add it back and restart. Only main difference I see is I'm not including `xdg-portals-hyprland` in the list of extraPortals. One other thing you could try is seeing if updating your lock file has any effect?

**Comment by George-Miao** [comment]
[Updated](https://github.com/George-Miao/nix-config/commit/d8e0c459c7c9ffe0c1d60d0f64077504b1826ace) lock, still no good unfortunately

**Comment by detroyejr** [comment]
I think I might have a fix. After experimenting a bit more, I noticed I have the same problem if I use `portal.xdgOpenUsePortal` in the xdg home manager module, but if I set it in the nixos module, the issue goes away. [They have identical options](https://github.com/nix-community/home-manager/blob/2f23fa308a7c067e52dfcc30a0758f47043ec176/modules/misc/xdg-portal.nix#L48) but there seems to be a conflict between the two implementations that's making something incompatible.

In other words try setting `xdg.portal.xdgOpenUsePortal` (and other relevant settings) in a nixos configuration.nix instead (like [this](https://github.com/George-Miao/nix-config/blob/183c94c75deda0544193d0b5f97f88ad6b940ac9/machine/Everest/default.nix) one?), reboot, and see if links work.
